commands/removerole.js

```diff
@@ -0,0 +1,142 @@
+            }
+            if (collected.first().content.toLowerCase() == "yes") {
+                return true
+            } else {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FFFF00")
+                    .addField("Information", "Command Cancelled")
+                ).then(m => m.delete(30000))
+                return false
+            }
+        }).catch(r => { console.log(r) })
+        if (!response) {
+            return
+        }
+        reaction_message_model.deleteOne({
+            messageID: message_id
+        }, (err, rem) => {
+            if (err) { console.log(err); return }
+            if (!rem) {
+                let noq = new Discord.RichEmbed()
+                    .setColor("#F0FFF0")
+                    .addField("Notification", "No roles were found for this message in the database to remove.")
+                return message.channel.send(noq).then(r => r.delete(30000))
+            } else {
+                let done = new Discord.RichEmbed()
+                    .setColor("#00FFF0")
+                    .addField("Notification", `All roles/emojis on message with the id **${message_id}** were deleted from the database.`)
+                return message.channel.send(done).then(r => r.delete(30000))
+            }
+        })
+    } else {
+        // remove only the specified emoji or corresponding role
+        reaction_message_model.findOne({
+            messageID: message_id
+        }, async (err, rem) => {
+            if (err) { console.log(err); return }
+            if (!rem) {
+                let noq = new Discord.RichEmbed()
+                    .setColor("#F0FFF0")
+                    .addField("Notification", "No roles were found in the database to remove.")
+                return message.channel.send(noq).then(r => r.delete(30000))
+            } else {
+                // temp = JSON.parse(JSON.stringify(rem.roles))
+                // remove the role from the roles object
+                var success = false
+                var reem
+                for (emoji in rem.roles) {
+                    if (args[1] == emoji || args[1] == rem.roles[emoji].name) {
+                        reem = emoji
+                        delete rem.roles[emoji]
+                        success = true
+                        // check if the roles object is now empty to remove the whole record
+                        if (hutil.isEmpty(rem.roles)) {
+                            reaction_message_model.deleteOne({
+                                messageID: message_id
+                            }, (err, rem2) => {
+                                if (err) { console.log(err); return }
+                            })
+                        } else {
+                            // if not empty just save the object after removing the role from it
+                            reaction_message_model.updateOne({ messageID: message_id }, { $set: { roles: rem.roles } }).catch(err => {
```

Comment:
This is a long lone, perhaps re-format?
```suggestion
						reaction_message_model
							.updateOne({ messageID: message_id }, { $set: { roles: rem.roles } })
							.catch(function(err) {
```